test.sh

```diff
@@ -145,10 +145,10 @@ assert 42 "int main() { return 42; }"
 assert 10 "int ret10() { return 10; } int main() { return ret10(); }"
 assert 0 "int main() { return 0; }"
 
-assert 42 "int identity(int x) { return x; } int main() { return identity(42); }"
 
-# assert 15 "int ADD(int a, int b) { return a + b; } int main() { return ADD(10, 5); }"
-# assert 21 "int ADD6(int a, int b, int c, int d, int e, int f) { return a + b + c + d + e + f; } int main() { return ADD6(1, 2, 3, 4, 5, 6); }"
+
+assert 15 "int ADD(int a, int b) { return a + b; } int main() { return ADD(10, 5); }"
+assert 21 "int ADD6(int a, int b, int c, int d, int e, int f) { return a + b + c + d + e + f; } int main() { return ADD6(1, 2, 3, 4, 5, 6); }"
 assert 42 "int identity(int x) { return x; } int main() { return identity(42); }"
```

Comment:
Add this test case and fix it by having each function have its own `var_map`.
`assert 4  "int main() {int x = 5; int y = 4; return id2(x, y);} int id1(int x, int y) { return x;} int id2(int y, int x) {return x;}"`
You can do this by changing your global `var_map` to be a map that returns a map. Then in codegen, the first key will be the function name, and the second key will be the variable name.

Reply:
```
ivan@ivan-server:~/repos/C-compiler-proj$ make test
clang++ -o leocc *.cpp -std=c++11 -g -fno-common '-Werror'
./test.sh 
Error: Variable 'x' is already declared.
```